1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -1,47 +1,79 @@
 class Solution(object):
-	# Given an n x n binary matrix grid, we return the shortest clear path in 
-	# the matrix, or -1 for no clear path.
+# Given an n x n binary matrix grid, we return the shortest clear path in 
+# the matrix, or -1 for no clear path.
 
-	# A clear path is a path from the top-left cell to the bottom right cell
-	# such that:
+# A clear path is a path from the top-left cell to the bottom right cell
+# such that:
 
-	 	# 1. all visited cells are 0
-	 	# 2. all adjacent cells are different and share an edge or corner
+	# 1. all visited cells are 0
+	# 2. all adjacent cells are different and share an edge or corner
 
+	def findAdjacent(grid: List[List[int]], start: Tuple[int, int]) -> List[Tuple[int, int]]:
+	# Create method to find surrounding cells that are clear path, given a  
+	# matrix and a current location.
 
-	def findAdjacent(grid: List[List[int]], start_row: Tuple[int, int], start_col: Tuple[int, int]) -> List[Tuple[int, int]]:
-		# Create method to find surrounding cells that are clear path, given a  
-		# matrix and a current location.
+		# Define directions as tuples (-1, 0), (1, 0), (0, -1), (0, 1), 
+		# (-1, -1), (-1, 1), (1, -1), (1, 1), (-1, -1).
+		directions = [(-1, 0), (1, 0), (0, -1), (0, 1), (-1, 1), (1, -1), (1, 1), (-1,-1)]
+		clearPaths = []
 
-			# Define directions as tuples (-1, 0), (1, 0), (0, -1), (0, 1), 
-			# (-1, -1), (-1, 1), (1, -1), (1, 1).
+		# For the given location, check if there are any 0's in the 8  
+		# surrounding spots. Also check if any of these are out of bounds.
+		for i in range(len(directions)):
```

Comment:
There are two common ways to use for. The way you picked gives you back an index. The other way is what you originally had, which gave you back the element: for direction in directions. The latter is generally preferable unless you need the index because you plan on looking at other elements. The reason is that working with an index can lead to all sorts of errors like array out of bounds or manipulating the list while iterating, whereas if you only have the direction object, you can't make these types of errors.

In your for loop, you only ever use i and you don't do any kind of index math (like looking at directions[i + 1] for example). This means that you just needed the tuple and not the index to the tuple.

This is the better way to work with the tuple:
for direction in directions:
    (xOffset, yOffset) = direction
    new_x = start[0] + xOffset
    ...

Reply:
I see! Very cool and I will keep that in mind for next time.